src/Workspaces/Core/Portable/Shared/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Extensions
+{
+    internal static class CommonParenthesizedExpressionSyntaxExtensions
+    {
+        public static bool IsSafeToChangeAssociativity(
+            this SyntaxNode parenthesizedExpression, SyntaxNode innerExpression, SemanticModel semanticModel)
+        {
+            // Now we'll perform a few semantic checks to determine whether removal 
+            // of the parentheses might break semantics. Note that we'll try and be 
+            // fairly conservative with these. For example, we'll assume that failing 
+            // any of these checks results in the parentheses being declared as necessary 
+            // -- even if they could be removed depending on whether the parenthesized
+            // expression appears on the left or right side of the parent binary expression.
+
+            // First, does the binary expression result in an operator overload being 
+            // called?
+            var symbolInfo = semanticModel.GetSymbolInfo(innerExpression);
+            if (symbolInfo.Symbol != null)
+            {
+                if (symbolInfo.Symbol is IMethodSymbol methodSymbol &&
+                    methodSymbol.MethodKind == MethodKind.UserDefinedOperator)
+                {
+                    return false;
+                }
+            }
+
+            // Second, check the type and converted type of the binary expression.
+            // Are they the same?
+            var innerTypeInfo = semanticModel.GetTypeInfo(innerExpression);
+            if (innerTypeInfo.Type != null && innerTypeInfo.ConvertedType != null)
+            {
+                if (!innerTypeInfo.Type.Equals(innerTypeInfo.ConvertedType))
```

Comment:
Perhaps consolidate the two `if`s

Reply:
Looks like this then:

```c#

            if (innerTypeInfo.Type != null &&
                innerTypeInfo.ConvertedType != null &&
                !innerTypeInfo.Type.Equals(innerTypeInfo.ConvertedType))
            {
                return false;
```

I think i prefer the code as it is now.